controls/models.py

```diff
@@ -326,9 +326,23 @@ def assign_baseline_controls(self, user, baselines_key, baseline_name):
 
     def statements(self, statement_type):
         """Return on the statements of statement_type produced by this element"""
+
         smts = Statement.objects.filter(producer_element = self, statement_type = statement_type)
         return smts
 
+    def consuming_systems(self):
+        """Return list of systems for which Element is producer_element of statement of type control_implementation"""
+
+        root_element_ids = set([ce['consumer_element'] for ce in Statement.objects.filter(producer_element=self).values('consumer_element').distinct()])
+        systems = [Element.objects.get(pk=root_element_id).system.all()[0] for root_element_id in root_element_ids if root_element_id]
```

Comment:
Change:
systems = Element.objects.filter(pk__in=root_element_ids)

Reply:
@alexanderward Looks like what I need is:
systems = System.objects.filter(pk__in=Element.objects.filter(pk__in=root_element_ids).values_list('system', flat=True))

Reply:
Resolved in 5ce09bc5